src/persistence/settings.cpp

```diff
@@ -2318,12 +2331,13 @@ bool Settings::getAutoLogin() const
     return autoLogin;
 }
 
-void Settings::setAutoLogin(bool state)
+void Settings::onSetAutoLogin(bool state)
 {
     QMutexLocker locker{&bigLock};
 
     if (state != autoLogin) {
         autoLogin = state;
+        saveGlobal();
```

Comment:
Because previously clicking the autologin checkbox called saveGlobal directly through Settings::getInstance(). The emit solution I wrote above could emit saveGlobal to preserve that behavior, but since all instances of autologin changing currently would behave the same, I opted to simply call the relevant function in its own class.

Reply:
Suppose to connect `autoLoginChanged` to `saveGlobal` and remove from here

Reply:
Absolutely. Changed as indicated.